core/lib/vm/src/tracers/traits.rs

```diff
@@ -88,10 +81,22 @@ pub enum TracerExecutionStatus {
     Stop(TracerExecutionStopReason),
 }
 
-/// Stop the vm execution if the tracer conditions are met
-pub trait ExecutionEndTracer<H: HistoryMode> {
-    // Returns whether the vm execution should stop.
-    fn should_stop_execution(&self) -> TracerExecutionStatus {
-        TracerExecutionStatus::Continue
+impl TracerExecutionStatus {
+    /// Chose the stricter ExecutionStatus
+    /// If both statuses are Continue, then the result is Continue
+    /// If one of the statuses is Abort, then the result is Abort
+    /// If one of the statuses is Finish, then the result is Finish
+    pub fn stricter(&self, other: &Self) -> Self {
```

Comment:
Nit: Couldn't `stricter` take `self` and `other` by value? IMO, this would be more idiomatic and would make cloning `reason` unnecessary.

Reply:
I've measured a performance and seems because it's kinda big structure the reference works better.